__tests__/unit/components/providers/ConnectionStatusProvider.test.tsx

```diff
@@ -0,0 +1,132 @@
+  const updateGoOnlineButtonHidden = () => updateGoOnlineButtonVisibility(false)
+  const updateInfoBannerVisible = () => updateOnlineInfoBannerVisibility(true)
+  const updateInfoBannerHidden = () => updateOnlineInfoBannerVisibility(false)
+  const resetInfoBanner = () => resetOnlineInfoBanner()
+
+  return (<>
+    <MemoryRouter initialEntries={['/currentUri']}>
+      <span data-testid="showButton">{String(showGoOnlineButton)}</span>
+      <span data-testid="isOffline">{String(isOffline())}</span>
+      <span data-testid="isInfoBannerVisible">{String(isOnlineInfoBannerVisible)}</span>
+      <button data-testid="update-connection-status-button--offline" onClick={updateToOffline}/>
+      <button data-testid="update-connection-status-button--online" onClick={updateToOnline}/>
+      <button data-testid="update-go-online-button--visible" onClick={updateGoOnlineButtonVisible}/>
+      <button data-testid="update-go-online-button--hidden" onClick={updateGoOnlineButtonHidden}/>
+      <button data-testid="update-info-banner--visible" onClick={updateInfoBannerVisible}/>
+      <button data-testid="update-info-banner--hidden" onClick={updateInfoBannerHidden}/>
+      <button data-testid="reset-info-banner" onClick={resetInfoBanner}/>
+    </MemoryRouter>
+  </>)
+}
+
+const renderTestComponent = () => render(<ConnectionStatusProvider><TestingComponent/></ConnectionStatusProvider>)
+
+describe('ConnectionStatusProvider', () => {
+  it('is show go online button false by default', () => {
+    const { queryByTestId } = renderTestComponent()
+    const testElement = queryByTestId('showButton')
+    expect(testElement!.innerHTML).toEqual('false')
+  })
+
+  it('is offline status true after setting to ConnectionStatus.OFFLINE', () => {
+    const { queryByTestId, getByTestId } = renderTestComponent()
+    fireEvent.click(getByTestId('update-connection-status-button--offline'))
+    const testElement = queryByTestId('isOffline')
+    expect(testElement!.innerHTML).toEqual('true')
+    const localStorageValue = localStorage.getItem(LocalStorageKey.ConnectionStatus)
+    expect(localStorageValue).toEqual(ConnectionStatus.OFFLINE)
+  })
+
+  it('offline status is false after setting to ConnectionStatus.ONLINE and online button visibility is false', () => {
+    const { queryByTestId, getByTestId } = renderTestComponent()
+    fireEvent.click(getByTestId('update-connection-status-button--online'))
+    const testElement = queryByTestId('isOffline')
+    expect(testElement!.innerHTML).toEqual('false')
+    const storedConnectionStatus = localStorage.getItem(LocalStorageKey.ConnectionStatus)
+    expect(storedConnectionStatus).toEqual(ConnectionStatus.ONLINE)
+    const goOnlineButton = queryByTestId('showButton')
+    expect(goOnlineButton!.innerHTML).toEqual('false')
+    const storedButtonVisibility = localStorage.getItem(LocalStorageKey.IsGoOnlineButtonVisible)
+    expect(storedButtonVisibility).toEqual('false')
```

Comment:
I don't mind this, but our other tests all have `expect` at the end (kind of Arrange-Act-Assert).

Reply:
Fixed in: https://github.com/gipfeli-io/gipfeli-frontend/pull/120